Modix.Bot/Behaviors/PromotionLoggingHandler.cs

```diff
@@ -37,37 +41,126 @@ public Task Handle(PromotionActionCreated notification, CancellationToken cancel
+        {
+            var promotionAction = await PromotionsService.GetPromotionActionSummaryAsync(promotionActionId);
+            var targetCampaign = promotionAction.Campaign ?? promotionAction.NewComment.Campaign;
+
+            var embed = new EmbedBuilder();
+
+            //Because we have comment creation as a separate operation from starting the campaign,
+            //we don't have access to the "initial" comment when a campaign is created. So, we have to
+            //note that a campaign was created, and actually send the log message when the first comment
+            //is created (containing the comment body)
+            switch (promotionAction.Type)
+            {
+                case PromotionActionType.CampaignCreated:
+
+                    _initialCommentQueue.TryAdd(targetCampaign.Id, embed
+                        .WithTitle("A new campaign has been started!")
+                        .AddField("If accepted, their new role will be", MentionUtils.MentionRole(targetCampaign.TargetRole.Id)));
+
+                    return null;
+                case PromotionActionType.CampaignClosed:
+
+                    var fullCampaign = (await PromotionsService.SearchCampaignsAsync(new PromotionCampaignSearchCriteria
+                    {
+                        Id = targetCampaign.Id
+                    }))
+                    .First();
+
+                    embed = embed
+                        .WithTitle("The campaign is over!")
+                        .AddField("Approval Rate", fullCampaign.GetApprovalPercentage().ToString("p"), true);
+
+                    var boldName = $"**{targetCampaign.Subject.Username}#{targetCampaign.Subject.Discriminator}**";
+                    var boldRole = $"**{MentionUtils.MentionRole(targetCampaign.TargetRole.Id)}**";
+
+                    switch (targetCampaign.Outcome)
+                    {
+                        case PromotionCampaignOutcome.Accepted:
+                            embed = embed
+                                .WithDescription($"Staff accepted the campaign, and {boldName} was promoted to {boldRole}! 🎉");
+                            break;
+                        case PromotionCampaignOutcome.Rejected:
+                            embed = embed
+                                .WithDescription($"Staff rejected the campaign to promote {boldName} to {boldRole}");
+                            break;
+                        case PromotionCampaignOutcome.Failed:
+                        default:
+                            embed = embed
+                                .WithDescription("There was an issue while accepting or denying the campaign. Ask staff for details.")
+                                .AddField("Target Role", MentionUtils.MentionRole(targetCampaign.TargetRole.Id), true);
+                            break;
+                    }
+
+                    break;
+                case PromotionActionType.CommentCreated:
```

Comment:
Do we still want this?

Reply:
This is a bit misleading, but it's necessary to capture the first comment that's created when the campaign starts (ie the "campaign reason"). If we don't want even that to show up, then sure, but I don't think anyone was opposed.